### PR TITLE
Python refactor reactor tests part 1

### PR DESCRIPTION
**Pull request by mambocab**
Initial refactoring for connection and timer tests in preparation for testing the `asyncio` reactor. Unless there's some strong objection, I'll merge this into my asyncio branch and write the tests on top of it.

### CODE REVIEW

**Comment by mambocab** [comment]
I've marked Alan and Jaume; feel free to pick just one of you to do the review.

File: tests/unit/io/test_twistedreactor.py [hunk 1]
```diff
@@ -17,17 +17,17 @@
...
 except ImportError:
-    twistedreactor = None  # NOQA
+    twistedreactor = TwistedConnection = None  # NOQA
+
 
 from cassandra.connection import _Frame
-from tests.unit.io.utils import submit_and_wait_for_completion, TimerCallback
 
 
 class TestTwistedTimer(unittest.TestCase):
```
Comment (bjmb) [inline tests/unit/io/test_twistedreactor.py]: This should inherit from `TimerTestMixin`?

**Comment by mambocab** [comment]
Yeah, I'll remove the commented stuff -- yesterday I thought it'd be reasonable to leave in my commented WIP stuff, but that's probably not a a good idea.

> This should inherit from TimerTestMixin?

:grimacing:

**Comment by mambocab** [comment]
Closed; IIRC this was used in the asyncio addition.

**Pull request closed without merging**